multiaccounts/accounts/database.go

```diff
@@ -129,6 +144,112 @@ func (db *Database) Close() error {
 	return db.db.Close()
 }
 
+func updateKeypairNameAndLastUsedIndex(tx *sql.Tx, keyUID string, keypairName string, index uint64) (err error) {
+	if tx == nil {
+		return errors.ErrDbTransactionIsNil
+	}
+
+	_, err = tx.Exec(`
+		UPDATE 
+			accounts 
+		SET 
+			keypair_name = ?,
+			last_used_derivation_index = ?
+		WHERE 
+			key_uid = ? 
+		AND
+			NOT chat`,
+		keypairName, index, keyUID)
+
+	return err
+}
+
+func updateAccountsTableIfNeeded(db *sql.DB) error {
```

Comment:
nit: this function name could probably have a better name e.g what to update and what is needed?

Reply:
fully agree, will update it later

Reply:
done